tests/testthat/test-model-integration.R

```diff
@@ -40,3 +40,127 @@ test_that("run_simulation with parameters = NULL", {
   expect_equal(length(output$timestep), 365)
 
 })
+
+test_that("run run_simulation twice using mockery_mock", {
+
+  # simple overrides
+  overrides <- list("pop" = list(1,3),
+                    "parameters" = list(1,3),
+                    "max_age" = list(1,3))
+
+  # get our run function that we will do some mocking to
+  run <- run_simulation_replicate
+
+  # mock the function to just return a data frame made up of our overrides
+  # to check they are being grabbed correctly
+  mockery::stub(
+    run,
+    'run_simulation',
+    function(pop, parameters, max_age) {
+      data.frame("timestep" = 1,
+                 "pop" = pop,
+                 "parameters" = parameters,
+                 "max_age" = max_age)
+    }
+  )
+
+  # run our mocked function
+  out <- run(repetitions = 2, overrides = overrides, parallel = FALSE)
+
+  # Now we can check the arguments are correctly filled from overrides
+  expect_true(out$repetition[1] == 1)
+  expect_true(out$timestep[2] == 1)
+  expect_true(out$pop[2] == 3)
```

Comment:
We would get better failure messages if we used [expect_equal](https://www.rdocumentation.org/packages/testthat/versions/0.11.0/topics/equivalence). And it would be a more robust test if we compared dataframes instead of just a bit of `out`. How about something like...



```suggestion
  expected <- data.frame(list(
    repetition= c(1, 2),
    timestep = c(1, 1),
    pop = pop,
    parameters = parameters,
    max_age = max_age # if we really expect max_age in our output
  ))
  expect_equal(out, expected)
```